redbot/cogs/mod/settings.py

```diff
@@ -101,16 +103,43 @@ class ModSettings(MixinMeta):
 
     @modset.command()
     @commands.guild_only()
-    async def deleterepeats(self, ctx: commands.Context):
-        """Enable auto-deletion of repeated messages."""
+    async def deleterepeats(self, ctx: commands.Context, repeats: int = None):
+        """Enable auto-deletion of repeated messages.
+
+        Must be between 2 and 20.
+
+        Set to -1 to disable this feature.
+        """
         guild = ctx.guild
-        cur_setting = await self.settings.guild(guild).delete_repeats()
-        if not cur_setting:
-            await self.settings.guild(guild).delete_repeats.set(True)
-            await ctx.send(_("Messages repeated up to 3 times will be deleted."))
+        if repeats is not None:
+            self.cache.pop(guild.id, None)  # remove cache with old repeat limits
```

Comment:
Last change request from me here.

Move this to after verifying a valid amount, and change it to create a new deque from the existing one.

for example:

```py
old_deque = deque([1,2,3], maxlen=4)
new_deque = deque(old_deque, maxlen=2)
```